rasa/core/channels/slack.py

```diff
@@ -387,6 +416,35 @@ def get_metadata(self, request: Request) -> Dict[Text, Any]:
 
         return {}
 
+    def is_request_from_slack_authentic(self, request: Request) -> bool:
+        """Validate a request from Slack for its authenticity.
+
+        Checks if the signature matches the one we expect from Slack. Ensures
+        we don't process request from a third-party disguising as slack.
+
+        Args:
+            request: incoming request to be checked
+
+        Returns:
+            `True` if the request came from Slack.
+        """
+        if self.slack_signing_secret is None:
+            # this is done for backwards compatibility, but will be removed in
+            # the next release
+            return True
+
+        slack_signing_secret = bytes(self.slack_signing_secret, "utf-8")
+
+        slack_signature = request.headers.get("X-Slack-Signature", "")
+        slack_request_timestamp = request.headers.get("X-Slack-Request-Timestamp", "")
```

Comment:
Should we also do step 2 of this guide https://api.slack.com/authentication/verifying-requests-from-slack#verifying-requests-from-slack-using-signing-secrets__a-recipe-for-security__step-by-step-walk-through-for-validating-a-request (avoid replay attacks)?

Reply:
very good point :+1: added this as a check (and some tests) 🚀